util/io.go

```diff
@@ -17,13 +17,37 @@
 package util
 
 import (
+	"errors"
 	"fmt"
 	"io"
 	"os"
 	"path/filepath"
 	"strings"
+	"syscall"
 )
 
+func MoveFile(src, dst string) error {
+	err := os.Rename(src, dst)
+	var errno syscall.Errno
+	if err != nil && errors.As(err, &errno) && errno == syscall.EXDEV {
+		// os.Rename() failed because src and dst are on different filesystems
+		// Must manually copy and delete the original file
+		return manuallyMoveFile(src, dst)
+	}
+	return err
+}
+
+func manuallyMoveFile(src, dst string) error {
+	tmpDst := dst + ".tmp"
+	if _, err := CopyFile(src, tmpDst); err != nil {
+		return err
+	}
+	if err := os.Rename(tmpDst, dst); err != nil {
+		return err
```

Comment:
Perhaps we should delete the tmpDst if this Rename fails?

Reply:
Added in d21751413